keps/sig-node/127-usernamespaces-support/README.md

```diff
@@ -0,0 +1,1122 @@
+-->
+
+The container runtime will have to be updated in the nodes to support this feature.
+
+The new `user` field in the `NamespaceOption` will be ignored by an old runtime
+without user namespaces support. The container will be placed in the host user
+namespace. It's a responsibility of the user to guarantee that a runtime
+supporting user namespaces is used when this feature is enabled.
+
+An old version of kubelet without user namespaces support can cause some
+issues too. In this case the runtime can wrongly infer that the `user` field
+is set to `POD` in the `NamespaceOption` message. To avoid this problem the
+runtime should check if the `mappings` field contains any mappings, an error
+should be raised otherwise.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md.
+
+The production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate
+    - Feature gate name: UserNamespacesSupport
+    - Components depending on the feature gate: kubelet
+
+* **Does enabling the feature change any default behavior?**
+  The default mode for usernamespaces is `Host`, so the default behaviour is not changed.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, by disabling the `UserNamespacesSupport` feature gate.
+  The effective user and group IDs of the process in the host would be different
+  before and after disabling the feature for pods running in `Cluster` and `Pod`
+  modes. This can cause access issues to pods accessing files saved in
+  volumes.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  The situation is very similar to the described above. The pod will be able to
+  access the files written when the feature was enabled but can have issues to
+  access those files written while the feature was disabled.
```

Comment:
Idem, I think all pods need to be re-created for the feature flag to take effect